command/connect/envoy/envoy.go

```diff
@@ -253,11 +260,26 @@ func (c *cmd) Run(args []string) int {
 	}
 
 	if c.register {
-		if !c.meshGateway {
-			c.UI.Error("Auto-Registration can only be used for mesh gateways")
+		c.isGateway = c.meshGateway || c.terminatingGateway
+		if !c.isGateway {
+			c.UI.Error("Auto-Registration can only be used for gateways")
+			return 1
+		}
+		if c.meshGateway && c.terminatingGateway {
+			c.UI.Error("More than one gateway type was specified")
```

Comment:
We could avoid this case by making the flag `-gateway=type`. As we add more types of gateways it may be easier to validate "x is a known type", instead of "these fields are all mutually exclusive". That might make the flags more intuitive to users since the invalid case can not exist.
We can translate the existing flag for mesh gateways into the new flag if it is set.


I noticed this in the RFC, but I wasn't sure I had enough context to comment at the time.

Reply:
Good catch! Doing that instead.